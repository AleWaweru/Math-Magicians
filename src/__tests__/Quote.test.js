import React from 'react';
import renderer from 'react-test-renderer';
import Quotes from '../components/quote';

describe('Quotes', () => {
  test('renders a loading spinner when loading', () => {
    const component = renderer.create(<Quotes isLoading />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders an error message when an error occurs', () => {
    const component = renderer.create(<Quotes hasError />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders a quote when quotes are available', () => {
    const quotes = [{ author: 'Alice', quote: 'Hello world!' }];
    const component = renderer.create(<Quotes quotes={quotes} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
