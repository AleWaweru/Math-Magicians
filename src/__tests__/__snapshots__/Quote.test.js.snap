// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes renders a loading spinner when loading 1`] = `
<div
  className="quotes-container"
>
  <h1>
    Quotes
  </h1>
  <ul />
</div>
`;

exports[`Quotes renders a quote when quotes are available 1`] = `
<div
  className="quotes-container"
>
  <h1>
    Quotes
  </h1>
  <ul />
</div>
`;

exports[`Quotes renders an error message when an error occurs 1`] = `
<div>
  Loading ...
</div>
`;
